From: Charles <charles05@canonical.com>
Date: Tue, 3 Mar 2026 08:11:26 +0000
Subject: Fallback to gexiv 0.14.2

gexiv is pinned to 0.14.x in Debian since 0.15.x introduced an API
change which would require a transition. That was not taken at the
time since nothing important depended on a newer version. Nautilus
50.alpha is now a dependant. It seems simple to continue using the
old version with no functional change in behaviour, so do that while
waiting for the transition.

Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/1111202
---
 .../image-properties/nautilus-image-properties-model.c  | 17 +++++++----------
 meson.build                                             |  2 +-
 2 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/extensions/image-properties/nautilus-image-properties-model.c b/extensions/image-properties/nautilus-image-properties-model.c
index ed10adb..7c0beed 100644
--- a/extensions/image-properties/nautilus-image-properties-model.c
+++ b/extensions/image-properties/nautilus-image-properties-model.c
@@ -109,7 +109,7 @@ static void
 append_gexiv_basic_info (NautilusImagesPropertiesModel *self)
 {
     const char *mime_type = gexiv2_metadata_get_mime_type (self->md);
-    GExiv2Orientation orientation = gexiv2_metadata_get_orientation (self->md, NULL);
+    GExiv2Orientation orientation = gexiv2_metadata_try_get_orientation (self->md, NULL);
     int width = gexiv2_metadata_get_pixel_width (self->md);
     int height = gexiv2_metadata_get_pixel_height (self->md);
 
@@ -178,15 +178,15 @@ append_gexiv2_tag (NautilusImagesPropertiesModel  *self,
 
     for (const char **i = tag_names; *i != NULL; i++)
     {
-        if (gexiv2_metadata_has_tag (self->md, *i, NULL))
+        if (gexiv2_metadata_try_has_tag (self->md, *i, NULL))
         {
-            g_autofree char *tag_value = gexiv2_metadata_get_tag_interpreted_string (self->md,
-                                                                                     *i,
-                                                                                     NULL);
+            g_autofree char *tag_value = gexiv2_metadata_try_get_tag_interpreted_string (self->md,
+                                                                                         *i,
+                                                                                         NULL);
 
             if (tag_description == NULL)
             {
-                tag_description = gexiv2_metadata_get_tag_description (*i, NULL);
+                tag_description = gexiv2_metadata_try_get_tag_description (*i, NULL);
             }
 
             /* don't add empty tags - try next one */
@@ -229,10 +229,7 @@ append_gexiv2_info (NautilusImagesPropertiesModel *self)
     append_gexiv2_tag (self, rights, _("Copyright"));
     append_gexiv2_tag (self, rating, _("Rating"));
 
-    gexiv2_metadata_get_gps_info (self->md, &longitude, &latitude, &altitude, NULL);
-
-    if (isnan (longitude) == 0 && isinf (longitude) == 0 &&
-        isnan (latitude) == 0 && isinf (latitude) == 0)
+    if (gexiv2_metadata_try_get_gps_info (self->md, &longitude, &latitude, &altitude, NULL))
     {
         g_autoptr (GString) gps_coords = g_string_new ("");
 
diff --git a/meson.build b/meson.build
index 9da9a87..62b24a0 100644
--- a/meson.build
+++ b/meson.build
@@ -127,7 +127,7 @@ dependency('wayland-client', required: gtk_wayland.found())
 selinux = dependency('libselinux', version: '>= 2.0', required: get_option('selinux'))
 cloudproviders = dependency('cloudproviders', version: '>= 0.3.1', required: get_option('cloudproviders'))
 if get_option('extensions')
-  gexiv = dependency('gexiv2-0.16', version: '>= 0.16.0')
+  gexiv = dependency('gexiv2', version: '>= 0.14.2')
   gdkpixbuf = dependency('gdk-pixbuf-2.0', version: '>= 2.30.0')
   gst_tag_dep = dependency('gstreamer-tag-1.0')
   gst_pbutils_dep = dependency('gstreamer-pbutils-1.0')
